be/src/exprs/vectorized/array_functions.h

```diff
@@ -40,6 +40,27 @@ class ArrayFunctions {
 
 #undef DEFINE_ARRAY_DISTINCT_FN
 
+#define DEFINE_ARRAY_INTERSECT_FN(NAME, PT)                                                     \
+    static ColumnPtr array_intersect_##NAME(FunctionContext* context, const Columns& columns) { \
+        return ArrayIntersect<PT>::process(context, columns);                                   \
+    }
+
+    DEFINE_ARRAY_INTERSECT_FN(boolean, PrimitiveType::TYPE_BOOLEAN);
```

Comment:
`APPLY_FOR_UNSIGN_OLAP_FIELD_TYPE` could be employed to reduce code for type enumeration.